src/lib.rs

```diff
@@ -141,7 +141,15 @@ impl Game {
         let state_ref = unsafe { &mut *self.state.get() };
 
         if state_ref.timing_mode != TimingMode::FrameSynchronized {
-            let n1 = (self.start_time.elapsed().as_nanos() - self.last_tick) as f64;
+            // Even with the non-monotonic Instant mitigation at the start of the event loop,
+            // there's still a theoretical chance of it not working.
+            // This check here should trigger if that happens and makes sure there's no panic from an underflow.
+            let mut elapsed = self.start_time.elapsed().as_nanos();
+            if elapsed < self.last_tick {
+                warn!("Elapsed time was less than last tick! elapsed: {}, last tick: {}", elapsed, self.last_tick);
```

Comment:
you might like to remove that logging, since it will happen quite often and unnecessarily keep spamming the console